bundles/org.openhab.binding.onkyo/src/main/java/org/openhab/binding/onkyo/internal/handler/OnkyoHandler.java

```diff
@@ -314,6 +334,30 @@ public void handleCommand(ChannelUID channelUID, Command command) {
         }
     }
 
+    private void populateInputs(NodeList selectorlist) {
+        List<StateOption> options = new ArrayList<>();
+
+        for (int i = 0; i < selectorlist.getLength(); i++) {
+            Element selectorItem = (Element) selectorlist.item(i);
+
+            options.add(new StateOption(String.valueOf(Integer.parseInt(selectorItem.getAttribute("id"), 16)),
+                    selectorItem.getAttribute("name")));
+        }
+        logger.debug("Got Input List from Receiver {}", options);
+
+        stateDescriptionProvider.setStateOptions(new ChannelUID(getThing().getUID(), CHANNEL_INPUT), options);
+        stateDescriptionProvider.setStateOptions(new ChannelUID(getThing().getUID(), CHANNEL_INPUTZONE2), options);
+        stateDescriptionProvider.setStateOptions(new ChannelUID(getThing().getUID(), CHANNEL_INPUTZONE3), options);
+    }
+
+    private State lastPowerState = OnOffType.OFF;
```

Comment:
Can you put all the fields at the top of the class?

Reply:
Sorry just saw this, have just fixed this up now.